File: Pwr/Sem4/JezykiIParadygmatyProgramowania/Lista6/zad2/gcd.pl
:- initialization(main, main).

egcd(A, 0, 1, 0, A).
egcd(A, B, X, Y, G) :-
    B > 0,
    R is A mod B,
    egcd(B, R, X1, Y1, G),
    X is Y1,
    Y is X1 - (A // B) * Y1.

run([A,B|_]) :-
    atom_number(A, A1),
    atom_number(B, B1),
    egcd(A1, B1, X, Y, G),
    write((X,Y,G)), nl.

main :-
    current_prolog_flag(argv, Argv),
    run(Argv),
    halt.

File: Pwr/Sem4/JezykiIParadygmatyProgramowania/Lista6/zad1/ocaml/gcd.ml
let rec egcd a b =
  if b = 0 then
    (1, 0, a)
  else
    let (x1, y1, g) = egcd b (a mod b) in
    let x = y1 in
    let y = x1 - (a / b) * y1 in
    (x, y, g)

let () =
  let a = int_of_string Sys.argv.(1) in
  let b = int_of_string Sys.argv.(2) in
  let (x, y, g) = egcd a b in
  Printf.printf "(%d, %d, %d)\n" x y g